resin/models/application.py

```diff
@@ -311,3 +313,152 @@ def disable_rolling_updates(self, app_id):
             'application', 'PATCH', params=params, data=data,
             endpoint=self.settings.get('pine_endpoint')
         )
+
+    def enable_device_urls(self, app_id):
+        """
+        Enable device urls for all devices that belong to an application
+
+        Args:
+            app_id (str): application id.
+
+        Returns:
+            OK/error.
+
+        Examples:
+            >> > resin.models.application.enable_device_urls('5685')
+            'OK'
+
+        """
+
+        params = {
+            'filter': 'belongs_to__application',
+            'eq': app_id
+        }
+        data = {
+            'is_web_accessible': True
+        }
+
+        return self.base_request.request(
+            'device', 'PATCH', params=params, data=data,
+            endpoint=self.settings.get('pine_endpoint')
+        )
+
+    def disable_device_urls(self, app_id):
+        """
+        Disable device urls for all devices that belong to an application.
+
+        Args:
+            app_id (str): application id.
+
+        Returns:
+            OK/error.
+
+        Examples:
+            >> > resin.models.application.disable_device_urls('5685')
+            'OK'
+
+        """
+
+        params = {
+            'filter': 'belongs_to__application',
+            'eq': app_id
+        }
+        data = {
+            'is_web_accessible': False
+        }
+
+        return self.base_request.request(
+            'device', 'PATCH', params=params, data=data,
+            endpoint=self.settings.get('pine_endpoint')
+        )
+
+    def grant_support_access(self, app_id, valid_period):
+        """
+        Grant support access to an application until a specified time.
+
+        Args:
+            app_id (str): application id.
+            valid_period (int): valid period in hour.
+
+        Returns:
+            OK/error.
+
+        Examples:
+            >> > resin.models.application.grant_support_access('5685', 2)
+            'OK'
+
+        """
+
+        if not valid_period or int(valid_period) <= 0:
+            raise exceptions.InvalidParameter('valid_period', valid_period)
+
+        expiry_timestamp = ((datetime.utcnow() - datetime.utcfromtimestamp(0)).total_seconds() + int(valid_period) * 3600) * 1000
```

Comment:
I just ran this code in a REPL, and it returns a float. Is the API okay with that, or should `expiry_timestamp` be converted to an `int`?

Reply:
@lurch yeah, the API is okay but it should be converted. However, I've changed the valid period to timestamp so this no longer is a prolem